### PR TITLE
Search performance optimisations #2

### PR DESCRIPTION
**Pull request by LotteHofstede**
This PR contains the same optimisations as #374 did, but now also for entity specific code on the search page.

### CODE REVIEW

File: src/app/+search-page/search-filters/search-filter/search-filter.service.ts [hunk 1]
```diff
@@ -14,15 +14,11 @@ import {
...
+import { RouteService } from '../../../core/services/route.service';
 import { SortDirection, SortOptions } from '../../../core/cache/models/sort-options.model';
 import { PaginationComponentOptions } from '../../../shared/pagination/pagination-component-options.model';
-import { SearchOptions } from '../../search-options.model';
-import { PaginatedSearchOptions } from '../../paginated-search-options.model';
 import { SearchFixedFilterService } from './search-fixed-filter.service';
 import { Params } from '@angular/router';
-import * as postcss from 'postcss';
-import prefix = postcss.vendor.prefix;
 // const spy = create();
```
Comment (atarix83) [inline src/app/+search-page/search-filters/search-filter/search-filter.service.ts]: this comment remained by mistake

File: src/app/+search-page/search-filters/search-filter/search-fixed-filter.service.spec.ts [hunk 2]
```diff
@@ -1,17 +1,15 @@
-import { SearchFixedFilterService } from './search-fixed-filter.service';
-import { RouteService } from '../../../shared/services/route.service';
-import { RequestService } from '../../../core/data/request.service';
-import { HALEndpointService } from '../../../core/shared/hal-endpoint.service';
-import { of as observableOf } from 'rxjs';
-import { RequestEntry } from '../../../core/data/request.reducer';
-import { FilteredDiscoveryQueryResponse, RestResponse } from '../../../core/cache/response.models';
+import {SearchFixedFilterService} from './search-fixed-filter.service';
```
Comment (atarix83) [inline src/app/+search-page/search-filters/search-filter/search-fixed-filter.service.spec.ts]: Could you revert spaces between curly brackets?

File: src/app/+search-page/search-labels/search-label/search-label.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { TranslateModule } from '@ngx-translate/core';
+import { ChangeDetectionStrategy, NO_ERRORS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { Observable, of as observableOf } from 'rxjs';
+import { Params } from '@angular/router';
+import {SearchLabelComponent} from './search-label.component';
```
Comment (atarix83) [inline src/app/+search-page/search-labels/search-label/search-label.component.spec.ts]: same as above, spaces missing

File: src/app/+search-page/search-labels/search-labels.component.spec.ts [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-import { SearchService } from '../search-service/search.service';
-import { ChangeDetectionStrategy, NO_ERRORS_SCHEMA } from '@angular/core';
-import { FormsModule } from '@angular/forms';
-import { SearchServiceStub } from '../../shared/testing/search-service-stub';
-import { Observable, of as observableOf } from 'rxjs';
-import { Params } from '@angular/router';
-import { ObjectKeysPipe } from '../../shared/utils/object-keys-pipe';
-import { SEARCH_CONFIG_SERVICE } from '../../+my-dspace-page/my-dspace-page.component';
-import { SearchConfigurationServiceStub } from '../../shared/testing/search-configuration-service-stub';
+import {SearchLabelsComponent} from './search-labels.component';
```
Comment (atarix83) [inline src/app/+search-page/search-labels/search-labels.component.spec.ts]: same as above, spaces missing

File: src/app/+search-page/search-filters/search-filter/search-filter.service.ts (hunk 1 above)
Comment (LotteHofstede) [reply src/app/+search-page/search-filters/search-filter/search-filter.service.ts]: Removed comment

File: src/app/+search-page/search-filters/search-filter/search-fixed-filter.service.spec.ts (hunk 2 above)
Comment (LotteHofstede) [reply src/app/+search-page/search-filters/search-filter/search-fixed-filter.service.spec.ts]: Reset original spacing

File: src/app/+search-page/search-labels/search-label/search-label.component.spec.ts (hunk 3 above)
Comment (LotteHofstede) [reply src/app/+search-page/search-labels/search-label/search-label.component.spec.ts]: Reset original spacing

File: src/app/+search-page/search-labels/search-labels.component.spec.ts (hunk 4 above)
Comment (LotteHofstede) [reply src/app/+search-page/search-labels/search-labels.component.spec.ts]: Reset original spacing

**Comment by LotteHofstede** [comment]
@tdonohue Thanks for noticing that.

This used to work in a previous version of this PR, because the `search/*` page used an endpoint to retrieve the correct query by the given filter name. When the filter didn't exist, the code wouldn't be able to retrieve a query, so it knew when the filter did not exist and could render a 404 page. 

In a [more recent PR](https://github.com/DSpace/dspace-angular/pull/442) that has already been merged, the fixed filters were replaced by the 
`configuration` parameter. This parameter is passed directly to the search endpoint and returns a 200 even if the configuration does not exist (it will simply ignore the parameter). This means I'm currently not able to determine whether the filter actually exists or not.

I will remove this feature from the PR description.

**Comment by tdonohue** [comment]
This PR needs rebasing, as it now has merge conflicts.

**Pull request merged**